tests/functional/metadata/mongodb/listLifecycleObject/current.spec.js

```diff
@@ -0,0 +1,414 @@
+
+                return done();
+            });
+        });
+    });
+
+    it('Should truncate list of current versions of objects', done => {
+        const params = {
+            listingType: 'DelimiterCurrent',
+            maxKeys: 2,
+        };
+        return metadata.listLifecycleObject(BUCKET_NAME, params, logger, (err, data) => {
+            assert.deepStrictEqual(err, null);
+            assert.strictEqual(data.IsTruncated, true);
+            assert.strictEqual(data.NextKeyMarker, 'pfx2-test-object');
+            assert.strictEqual(data.Contents.length, 2);
+            assert.strictEqual(data.Contents[0].key, 'pfx1-test-object');
+            assert.strictEqual(data.Contents[1].key, 'pfx2-test-object');
+            return done();
+        });
+    });
+
+    it('Should list the following current versions of objects', done => {
+        const params = {
+            listingType: 'DelimiterCurrent',
+            marker: 'pfx2-test-object',
+        };
+        return metadata.listLifecycleObject(BUCKET_NAME, params, logger, (err, data) => {
+            assert.deepStrictEqual(err, null);
+            assert.strictEqual(data.IsTruncated, false);
+            assert.strictEqual(data.Contents.length, 1);
+            assert.strictEqual(data.Contents[0].key, 'pfx3-test-object');
+            return done();
+        });
+    });
+
+    it('Should list current versions that start with prefix', done => {
+        const params = {
+            listingType: 'DelimiterCurrent',
+            prefix: 'pfx2',
+        };
+        return metadata.listLifecycleObject(BUCKET_NAME, params, logger, (err, data) => {
+            assert.deepStrictEqual(err, null);
+            assert.strictEqual(data.IsTruncated, false);
+            assert.strictEqual(data.Contents.length, 1);
+            assert.strictEqual(data.Contents[0].key, 'pfx2-test-object');
+
+            return done();
+        });
+    });
+
+    it('Should return the list of current versions modified before 1970-01-01T00:00:01.010Z with prefix pfx1', done => {
+        const params = {
+            listingType: 'DelimiterCurrent',
+            beforeDate: '1970-01-01T00:00:01.010Z',
+            maxKeys: 1,
+            prefix: 'pfx1',
+        };
+        return metadata.listLifecycleObject(BUCKET_NAME, params, logger, (err, data) => {
+            assert.deepStrictEqual(err, null);
+            assert.strictEqual(data.IsTruncated, false);
+            assert.strictEqual(data.Contents.length, 1);
+            assert.strictEqual(data.Contents[0].key, 'pfx1-test-object');
+
+            return done();
+        });
+    });
+
+    it('Should return an Internal error when bucket non existent (current)', done => {
+        const bucketName = 'non-existent-bucket';
+        const params = {
+            listingType: 'DelimiterCurrent',
+        };
+        return metadata.listLifecycleObject(bucketName, params, logger, (err, data) => {
+            expect(err.InternalError).toBe(true);
```

Comment:
We should probably stick to using `assert` everywhere, unless we use `expect` consistenly in the whole file.

Reply:
Do you want to address this? I personally don't think it's a big deal but you may have forgotten ;)